src/libstd/collections/hash/table.rs

```diff
@@ -801,25 +784,31 @@ impl<K, V> RawTable<K, V> {
         let hashes = buffer.offset(hash_offset as isize) as *mut HashUint;
 
         RawTable {
-            capacity: capacity,
+            capacity_mask: capacity.wrapping_sub(1),
             size: 0,
             hashes: TaggedHashUintPtr::new(hashes),
             marker: marker::PhantomData,
         }
     }
 
-    fn first_bucket_raw(&self) -> RawBucket<K, V> {
-        let hashes_size = self.capacity * size_of::<HashUint>();
-        let pairs_size = self.capacity * size_of::<(K, V)>();
+    fn raw_bucket(&self) -> RawBucket<K, V> {
```

Comment:
This seems like a worse name, since it isn't clear which raw bucket is returned. Maybe `first_raw_bucket` or `raw_bucket_first` or stick with the original `first_bucket_raw` (presumably paired with `bucket_at_raw`).